File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        int profit  = 0 ;
        int  min  =  prices[0] ;
        for(int  i = 1 ; i<prices.size();i++)
        {
            if(min<prices[i])
            {
                profit = max(profit ,prices[i]-min) ;
            }
            else 
          {
            min  = prices[i] ;
          }
    
        
        }

        return profit ;
    }
};

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        i  = 0 
        j = 0
        while i<len(arr) :
            Max = 0
            j=i 
            while j < len(arr) -1:
                if arr[j+1] > Max:
                    Max =  arr [j+1]
                j +=1 
            arr[i] = Max
            i  +=1
        if i>0:
            arr[i-1]  = -1
        return arr 
